operations.js

```diff
@@ -0,0 +1,223 @@
+const utils = require('./utils');
+const db = require('@splinterlands/pg-querybuilder');
+const interface = require('@splinterlands/hive-interface');
+
+let hive = null;
+
+let _options = {
+	logging_level: 3,
+	rpc_error_limit: 20,
+	rpc_nodes: ["https://api.hive.blog", "https://anyx.io", "https://hived.splinterlands.com"],
+	ws_url: "http://localhost:1234",
+	state_file_name: 'sl-state.json',
+	game_api_url: 'http://localhost:3000',
+	prefix: "dev-sm_"
+};
+
+function init(options) {
+	_options = Object.assign(_options, options);
+	utils.set_options(_options);
+	hive = new interface.Hive(_options);
+
+	if(_options.connection)
+		db.init({ connection: _options.connection });
+}
+
+async function processPurchase(purchase_id, payment) {
+	return new Promise(async (resolve, reject) => {
+		let purchase = await db.lookupSingle('purchases', { uid: purchase_id });
+
+		if(!purchase)
+			return reject({ error: `Purchase [${purchase_id}] not found.` });
+
+		let amount = parseFloat(payment);
+		let currency = utils.getCurrency(payment);
+
+		if(currency != purchase.currency)
+			return reject({ error: `Invalid currency sent. Expected ${purchase.currency} but got ${currency}.` });
+
+		// Make sure the payment amount is enough!
+		if(amount * 1.02 < parseFloat(purchase.ext_currency_amount))
```

Comment:
Consider using Joi to validate the purchase object at the beginning. Lots of examples of Joi in the Steemmonsters repo.